src/app/shared/search/filters/search-filters.component.ts

```diff
@@ -37,27 +40,50 @@ export class SearchFiltersComponent implements OnInit {
   formGroup: FormGroup = new FormGroup({});
 
   constructor(private searchService: SearchService,
-              private orderService: OrderService,
-              private jurisdictionService: JurisdictionService) { }
+    private orderService: OrderService,
+    private jurisdictionService: JurisdictionService,
+    private windowService: WindowService,
+    private platformLocation: PlatformLocation) {
+  }
 
   ngOnInit(): void {
     this.selected = {};
-    if (this.jurisdictions.length === 1) {
+    const jurisdiction = this.windowService.getLocalStorage('search-jurisdiction');
+    if (this.jurisdictions.length === 1 || jurisdiction) {
       this.selected.jurisdiction = this.jurisdictions[0];
+      if (jurisdiction) {
+        const localStorageJurisdiction = JSON.parse(jurisdiction);
+        this.selected.jurisdiction = this.jurisdictions.filter(j => j.id === localStorageJurisdiction.id)[0];
+      }
       this.onJurisdictionIdChange();
     }
+
     if (this.autoApply === true) {
-      this.apply();
+      this.selected.formGroup = this.formGroup;
+      this.selected.page = 1;
+      this.selected.metadataFields = this.getMetadataFields();
+      this.onApply.emit(this.selected);
     }
   }
 
   apply(): void {
     this.selected.formGroup = this.formGroup;
     this.selected.page = 1;
     this.selected.metadataFields = this.getMetadataFields();
+    this.populateValuesInLocalStorage();
     this.onApply.emit(this.selected);
   }
 
+  populateValuesInLocalStorage(): void {
+    this.windowService.setLocalStorage('search-form-group-value',
+      JSON.stringify(this.selected.formGroup.value));
+    this.windowService.setLocalStorage('search-metadata-fields', JSON.stringify(this.selected.metadataFields));
+    this.windowService.setLocalStorage('search-jurisdiction', JSON.stringify(this.selected.jurisdiction));
+    if (this.selected.caseType) {
+      this.windowService.setLocalStorage('search-caseType', JSON.stringify(this.selected.caseType))
+    }
```

Comment:
Instead of setting different key/value pairs, wouldn't it be easier to have an object:
```js
{
    'search-form-group-value': ...,
    'search-metadata-fields': ...,
    'search-jurisdiction': ...,
    'search-caseType': ...
}
```

And stringify it all in one go?